vault/cluster/cluster.go

```diff
@@ -33,14 +33,23 @@ type Client interface {
 // off a connection for a cluster listener application.
 type Handler interface {
 	ServerLookup(context.Context, *tls.ClientHelloInfo) (*tls.Certificate, error)
-	CALookup(context.Context) (*x509.Certificate, error)
+	CALookup(context.Context) ([]*x509.Certificate, error)
 
 	// Handoff is used to pass the connection lifetime off to
 	// the handler
 	Handoff(context.Context, *sync.WaitGroup, chan struct{}, *tls.Conn) error
 	Stop() error
 }
 
+type ClusterHook interface {
```

Comment:
Is there a reason this needs to be repeated here? Why not use the version that's in physical/rafttypes.go? (Or since that calls into here, why not have the physical side use this ClusterHook interface instead of that one?)